vnext/Shared/HermesSamplingProfiler.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include <chrono>
+
+#include <hermes/hermes.h>
+#include <winrt/Windows.Storage.h>
+#include "Unicode.h"
+
+#include "HermesSamplingProfiler.h"
+
+namespace Microsoft::ReactNative {
+
+namespace {
+std::future<std::string> getTraceFilePath() {
```

Comment:
@vmoroz what is the difference in behavior with `std::future` vs winrt `IAsyncOperation`? I remember hearing about a preference before but do not recall details.

Reply:
@vmoroz Vladimir, Please share if you have any inputs here ..